### PR TITLE
Get the upstream name from the pushed virtual branch

### PR DESCRIPTION
**Pull request by estib-vega**
Once the push of a virtual branch succeeded, return the ref name and the remote name in order to correctly determine which branch name to create a PR from.

### CODE REVIEW

**Comment by ukarlsson** [comment]
👍

File: apps/desktop/src/lib/utils/branch.ts [hunk 1]
```diff
@@ -4,13 +4,17 @@ const PREFERRED_REMOTE = 'origin';
...
 	}
 
-	const parts = ref.split(BRANCH_SEPARATOR);
-	return parts.length > 1 ? parts.slice(1).join(BRANCH_SEPARATOR) : ref;
+	const originPrefix = `${remote}${BRANCH_SEPARATOR}`;
+	if (ref.startsWith(originPrefix)) {
+		ref = ref.replace(originPrefix, '');
+	}
+
+	return ref;
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/utils/branch.ts]:
```suggestion
export function getBranchNameFromRef(ref: string, remote?: string): string | undefined {
 	if (remote && ref.startsWith(REF_REMOTES_PREFIX)) {
 		ref = ref.replace(REF_REMOTES_PREFIX, '');
 		
 		if (!ref.startsWith(originPrefix)) {
 		  throw new Error("Failed to parse branch name due to incorrect remote provided");
    }

    ref = ref.replace(originPrefix, '');
 	}

 	const originPrefix = `${remote}${BRANCH_SEPARATOR}`;


 	return ref;
}
```

**Review by Caleb-T-Owens — CHANGES_REQUESTED** [review]
It would be nice if it failed hard if you're trying to parse a remote branch but you give it a non-matching remote

**Comment by estib-vega** [comment]
> It would be nice if it failed hard if you're trying to parse a remote branch but you give it a non-matching remote

@Caleb-T-Owens Done 👍

File: apps/desktop/src/lib/utils/branch.test.ts [hunk 2]
```diff
@@ -4,20 +4,30 @@ import { expect, test, describe } from 'vitest';
 describe.concurrent('getBranchNameFromRef', () => {
 	test('When provided a ref with a remote prefix, it returns the branch name', () => {
 		const ref = 'refs/remotes/origin/main';
+		const remote = 'origin';
 
-		expect(BranchUtils.getBranchNameFromRef(ref)).toBe('main');
+		expect(BranchUtils.getBranchNameFromRef(ref, remote)).toBe('main');
 	});
 
 	test('When provided a ref without a remote prefix, it returns the branch name', () => {
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/utils/branch.test.ts]: This test name is no longer accurate

Comment (estib-vega) [reply apps/desktop/src/lib/utils/branch.test.ts]: Sorry about that! Fixed in the latest version

**Review by Caleb-T-Owens — APPROVED** [review]
LGTM

**Pull request merged**